tests/models/test_default_evaluator.py

```diff
@@ -65,15 +68,91 @@ def assert_dict_equal(d1, d2, rtol):
         assert np.isclose(d1[k], d2[k], rtol=rtol)
 
 
-def test_regressor_evaluation(linear_regressor_model_uri, diabetes_dataset):
+def evaluate_model_helper(
+    model,
+    baseline_model,
+    data,
+    targets,
+    model_type: str,
+    dataset_name=None,
+    evaluators=None,
+    evaluator_config=None,
+    eval_baseline_model_only=False,
+):
+
+    """
+    Helper function for testing MLflow.evaluate
+    To test if evaluation for baseline model does not log metrics and artifacts;
+    we set "is_baseline_model" to true for all evaluator_config
+    """
+    if eval_baseline_model_only:
+        if not evaluator_config:
+            evaluator_config = {"is_baseline_model": True}
+        elif not evaluators or evaluators == "default":
+            evaluator_config.update({"is_baseline_model": True})
+        else:
+            for config in evaluator_config.values():
+                config.update({"is_baseline_model": True})
+
+    return evaluate(
+        model=baseline_model if eval_baseline_model_only else model,
```

Comment:
Why would the baseline_model be passed into this parameter? I'm confused.

Reply:
For testing if metrics and artifacts are not logged for evaluation of baseline model, it is hard to test that if we pass in both model and baseline_model, because it is hard to tell whether the baseline_model logs some metrics that overwritten the ones model generated, so I came up with this to test if we pass in a baseline model and only evaluate it without candidate model to verify that nothing is logged and no artifacts is generated so that we know the boolean flagis_baseline_model actually disable logging metrics and artifacts. To do that I have to pass the boolean flag to DefaultEvaluator which is through the evaluator config and evaluate only baseline model without candidate model to see if there is nothing logged.

Reply:
This makes a lot more sense to me now. Would it be possible to pass in some dummy model here if `eval_baseline_model_only`? Otherwise, we can add a comment to explain what this line is doing.

Reply:
Cause different dataset need different type of model, I will add a comment to explain this line.

Reply:
Now that you are actually evaluating the `baseline_model` now as opposed to the `model`, can you update this conditional to pass None when `eval_baseline_model_only` is `True`?